pkgs/development/libraries/qt-5/5.11/qtwebengine-darwin-sdk-10.10.patch

```diff
@@ -0,0 +1,160 @@
+Fix build against 10.10 SDK
+
+The SecKey part perhaps could be fixed by implementing a revert to
+https://chromium.googlesource.com/chromium/src.git/+/8418e098b9cbedf884878b61dcd3292c515845cf%5E%21/#F0
+
+--- a/src/3rdparty/chromium/content/browser/renderer_host/input/web_input_event_builders_mac.mm
++++ b/src/3rdparty/chromium/content/browser/renderer_host/input/web_input_event_builders_mac.mm
+@@ -1,3 +1,4 @@
++#define NSEventTypeScrollWheel 22
+ // Copyright 2015 The Chromium Authors. All rights reserved.
+ // Use of this source code is governed by a BSD-style license that can be
+ // found in the LICENSE file.
+--- a/src/3rdparty/chromium/net/ssl/ssl_platform_key_mac.cc
++++ b/src/3rdparty/chromium/net/ssl/ssl_platform_key_mac.cc
+@@ -48,21 +48,6 @@ namespace net {
+ 
+ namespace {
+ 
+-// TODO(davidben): Remove this when we switch to building to the 10.13
```

Comment:
This patch looks a bit scary. What is it doing?

Reply:
In short, this disables signing using "Certificate, Key, and Trust Services API" and makes it use CSSM API instead. If you look below, this branch, that I removed, is hidden behind a version check:
```c++
if (base::mac::IsAtLeastOS10_12()) {
  return /* bla using SecKey */;
}

return /* bla using CSSM */;
```

So I only drop an optional execution path.

 There is a nice overview of API's in http://www.arp.com/medias/13902065.pdf . From their explanation I understood that CSDA (and CSSM) are more sophisticated API's produced by some open source standard group while "Certificate, Key, and Trust Services API" is what Apple wants us to use. They seem to allude that latter is just a wrapper around former:

> Where possible, you should use one of the following instead of using CDSA directly:
> * The Certificate, Key, and Trust Services API for general encryption, key management, and other tasks.

I suppose the reason CSSM is not a preferred API is to maintain chromium in a future proof state. This is my speculation of course. My best understanding is that this patch is harmless and we can drop it once 10.12 SDK lands.

Reply:
ok.